Server/JokesOnYou.Web.Api/Services/UserService.cs

```diff
@@ -37,5 +44,40 @@ public async Task UpdateUser(UserUpdateDTO updateDTO)
         {
             throw new NotImplementedException();
         }
+
+        public async Task<UserReplyDTO> LoginUser(UserLoginDTO userLogin)
+        {
+            var user = await _userRepository.GetUserByEmail(userLogin.Email);
+            if (user != null)
+            {
+                var signInResult = await _signInManager.CheckPasswordSignInAsync(user, userLogin.Password, false);
+
+                if (signInResult.Succeeded)
+                {
+                    var userReplyDTO = new UserReplyDTO()
+                    {
+                        Id = user.Id,
+                        Email = user.Email,
+                        UserName = user.Email,
+                        Role = "",
+                        Token = _tokenService.GetToken(user)
+                    };
+                    return userReplyDTO;
+                }
+                else
+                {
+                    throw new NotImplementedException("Sign in failed");
```

Comment:
Return Unautorized instead?

Reply:
> Return Unautorized instead?

since hes inside the service he cant return unauthorized but he can throw an unauthorizedexception and have the global exc handler catch it